lib/improver/utilities/cube_metadata.py

```diff
@@ -145,11 +145,23 @@ def add_coord(cube, coord_name, changes, warnings_on=False):
         bounds = changes['bounds']
     units = None
     if 'units' in changes:
-        units = changes["units"]
-    new_coord = new_coord_method(long_name=coord_name,
-                                 points=points,
-                                 bounds=bounds,
-                                 units=units)
+        units = changes['units']
+    var_name = None
+    if 'var_name' in changes:
+        var_name = changes['var_name']
+
+    try:
+        new_coord = new_coord_method(
+            standard_name=coord_name, var_name=var_name, points=points,
+            bounds=bounds, units=units)
+    except ValueError as cause:
+        if 'is not a valid standard_name' in str(cause):
+            new_coord = new_coord_method(
+                long_name=coord_name, var_name=var_name, points=points,
+                bounds=bounds, units=units)
+        else:
+            raise ValueError(cause)
```

Comment:
I'm not sure you've got a test that goes into this else statement?

Reply:
Time for more cute animals in unit tests...